### DISCUSSION TITLE
Wrapper for validation errors

### DISCUSSION THREAD

**Discussion by Augen2203** [spatie/laravel-data#258] [Q&A]
Greetings.
I wanted to clarify whether it is possible to set a wrapper for an array with validation errors, since sometimes we are faced with the following situation:

From the form we receive an array of data of the following format:
`['category' => ['name' = '....', ...]]`

In this case, we skip the default Laravel validation class as stated in the documentation and validate directly with the Data class.
`CategoryData::validateAndCreate($request->get('category'));`

With this form format, we cannot separately select a field that has not passed validation, since in the error array, field keys are already stored without a wrapper in which they were received from the form

We get:
<img width="360" alt="Знімок екрана 2022-11-18 о 19 45 11" src="https://user-images.githubusercontent.com/59070939/202768875-5753b65c-a49e-495a-a75f-d695d3d9cba0.png">

The way I would like to receive it:
<img width="343" alt="Знімок екрана 2022-11-18 о 19 45 16" src="https://user-images.githubusercontent.com/59070939/202769111-e3b8ead2-3591-44b4-8ea4-0b7095cede26.png">

The result that I would like to get can be achieved, for example, by catching the validation exception and generating the data yourself, like this:

```
try {
   ...
} catch (ValidationException $e) {
    return redirect()->back()->withErrors(Arr::prependKeysWith($e->errors(), 'category.'));
}
```

But I'm interested in a more elegant solution, perhaps it is already provided in the package, or is planned for implementation.

Thanks

**Comment by rubenvanassche** [comment]
Hmm this feels a bit like a too niche feature, I don't think we're going to implement such a thing. You probably could create a custom validation Rule within Laravel which does this for you. It would then internally call `CategoryData::validate`.
